userapi/src/test/java/fk/prof/userapi/api/MockAggregatedProfile.java

```diff
@@ -0,0 +1,419 @@
+    }
+
+    private List<AggregatedProfileModel.FrameNodeList> buildFrameNodes(String path) throws Exception {
+        FileInputStream fis = new FileInputStream(path);
+        List<AggregatedProfileModel.FrameNodeList> list = new ArrayList<>();
+
+        int count = 0;
+
+        while(fis.available() != -1) {
+            AggregatedProfileModel.FrameNodeList nodelist = AggregatedProfileModel.FrameNodeList.parseDelimitedFrom(fis);
+            if(nodelist == null) {
+                break;
+            }
+            count +=  nodelist.getFrameNodesCount();
+            list.add(nodelist);
+        }
+
+        fis.close();
+
+        System.out.println(list.size());
+        System.out.println(count);
+
+        return list;
+    }
+
+    private Map<Long, FinalizedProfileWorkInfo> buildProfilesWorkInfo(LocalDateTime aggregationStart, int count1, int count2) {
+
+        AggregatedProfileModel.RecorderInfo r1 = AggregatedProfileModel.RecorderInfo.newBuilder()
+                .setIp("192.168.1.1")
+                .setHostname("some-box-1")
+                .setAppId("app1")
+                .setInstanceGroup("ig1")
+                .setCluster("cluster1")
+                .setInstanceId("instance1")
+                .setProcessName("svc1")
+                .setVmId("vm1")
+                .setZone("chennai-1")
+                .setInstanceType("c1.xlarge").build();
+
+        AggregatedProfileModel.RecorderInfo r2 = AggregatedProfileModel.RecorderInfo.newBuilder()
+                .setIp("192.168.1.2")
+                .setHostname("some-box-2")
+                .setAppId("app1")
+                .setInstanceGroup("ig1")
+                .setCluster("cluster1")
+                .setInstanceId("instance2")
+                .setProcessName("svc1")
+                .setVmId("vm2")
+                .setZone("chennai-1")
+                .setInstanceType("c1.xlarge").build();
+
+        FinalizedProfileWorkInfo wi = new FinalizedProfileWorkInfo(1, r1, AggregationState.COMPLETED, aggregationStart.plusSeconds(10), aggregationStart.plusSeconds(10 + 60), 60,
+                buildMap("full-app-trace", 5),
+                buildMap(AggregatedProfileModel.WorkType.cpu_sample_work, count1)
+                );
+
+        FinalizedProfileWorkInfo wi2 = new FinalizedProfileWorkInfo(1, r2, AggregationState.RETRIED, aggregationStart.plusSeconds(24), aggregationStart.plusSeconds(24 + 60), 60,
+                buildMap("full-app-trace", 10),
+                buildMap(AggregatedProfileModel.WorkType.cpu_sample_work, count2)
+                );
+
+        return buildMap(101l, wi, 102l, wi2);
+    }
+
+    // A temp byteBuffer is reused everytime to write a abstractMessage. This avoids the creation of temp buffer everytime
+    // you want to write the message but its content is copied to the outputStream.
+    public void testCodedOutputStream_firstWriteToByteBufferThenCopyToOutputStream() throws Exception {
```

Comment:
There are multiple strategies to write proto buf to a stream. These are present for reference.